src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.SetupPackage.vsmanproj

```
@@ -38,19 +38,26 @@
   <ItemGroup>
     <_PackageStubFiles Include="$(ComponentIntermediateOutputPath)*.stub"/>
     <MergeManifest Include="@(_PackageStubFiles->'$(SetupOutputPath)%(Filename).json')"/>
+    <MergeManifest Include="@(_PackageStubFiles->'$(SetupOutputPath)%(Filename).json')">
+      <SBOMFileLocation>$(BUILD_SOURCESDIRECTORY)\artifacts\sbom\spdx_2.2\manifest.spdx.json</SBOMFileLocation>
+      <SBOMFileDestPath>$(SetupOutputPath)</SBOMFileDestPath>
+    </MergeManifest>
   </ItemGroup>
 
-  <Import Project="$(SwixBuildPath)build\MicroBuild.Plugins.*.targets" />
+  <Import Project="$(SwixBuildPath)build\Microsoft.VisualStudioEng.MicroBuild.Plugins.SwixBuild.targets" />
+
+  <Target Name="_BuildManifest" DependsOnTargets="_SetManifestBuildVersion;GenerateSetupManifest;AddSBOM" />
 
-  <Target Name="_BuildManifest" DependsOnTargets="_SetManifestBuildVersion;GenerateSetupManifest" />
-  
   <!--
     Each stub file contains VSIX version of the respective VSIX file.
     We require that all VSIXes included in a single VS insertion component have the same version.
     This version will be set to ManifestBuildVersion.
   -->
   <Target Name="_SetManifestBuildVersion">
 
+    <Error Condition="!Exists('$(BUILD_SOURCESDIRECTORY)\artifacts\sbom\spdx_2.2\manifest.spdx.json')" 
```

Should this be checking for the existence of SBOMFileLocation. I assume that's what AddSBOM wants?

---

I added this here, there was no logs if the sboms were getting included initially. 

Then we want this to fail if there are no sboms getting generated, else the check in vs insertion will fail at a the later stage.. I feel it will be good to catch this error here... I can change this to warning atleast.. 

AddSBOM task does not log anything, so this error might get swallowed..

---

Also AddSbom task is used to add SBOM link to vsmanifest.. It does not check if the SBOM is created or not..

---

AddSbom is no longer required now. 

The reason it did not call the task earlier was it was not able to find the sbom and there was no log that mentioned that. After extensive testing I came to the conclusion after setting the recognizable path for sbom, I am able to generate vsmanifest